crates/src/rating/elo.cairo

```diff
@@ -0,0 +1,174 @@
+//! Elo struct and rating methods.
+//! Source: https://github.com/saucepoint/elo-lib/blob/main/src/Elo.sol
+
+// Core imports
+
+use core::integer::{u32_sqrt, u64_sqrt, u128_sqrt, u256_sqrt};
+
+// Constants
+
+const MULTIPLIER: u32 = 10_000;
+const SCALER: u256 = 800;
+
+// Errors
+
+mod errors {
+    const ELO_DIFFERENCE_TOO_LARGE: felt252 = 'Elo: difference too large';
+}
+
+/// Elo implementation..
+#[generate_trait]
+impl EloImpl of EloTrait {
+    /// Calculates the change in ELO rating, after a given outcome.
+    /// # Arguments
+    /// * `rating_a` - The ELO rating of the player A.
+    /// * `rating_b` - The ELO rating of the player B.
+    /// * `score` - The score of the player A, scaled by 100. 100 = win, 50 = draw, 0 = loss.
+    /// * `k` - The k-factor or development multiplier used to calculate the change in ELO rating. 20 is the typical value.
+    /// # Returns
+    /// * `change` - The change in ELO rating of player A.
```

Comment:
Can we used a signed int here?

Reply:
Since signed integers have no Div trait implemented it will just make things more complex on our side imho.

Issue opened: https://github.com/starkware-libs/cairo/issues/5450
Old PR attempt: https://github.com/starkware-libs/cairo/pull/4342

Reply:
Ah ok my bad I didn't realize that